csm_api_client/service/cfs.py

```diff
@@ -860,7 +860,20 @@ def _update_container_status(self) -> None:
         # Update both init_container_status_by_name and container_status_by_name
         for prefix in ('init_', ''):
             status_by_name = getattr(self, f'{prefix}container_status_by_name')
-            for container_status in getattr(self.pod.status, f'{prefix}container_statuses'):
+            if self.pod.status.init_container_statuses is None and self.pod.status.container_statuses is None:
+                LOGGER.info("Both init_container_statuses and container_statuses are None. Waiting for status update.")
+                return
```

Comment:
We don't need this check anymore. We will just check init_container_statuses on the first iteration of the loop on line 861 and then container_statuses on the second iteration. When either of those is `None`, we will exit the loop and avoid trying to iterate over a `NoneType`.

```suggestion
```

Reply:
Addressed